frontends/election-manager/src/components/new_print_button.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useContext, useState, useCallback } from 'react';
+
+import {
+  Button,
+  StyledButtonProps,
+  Modal,
+  useCancelablePromise,
+  useMountedState,
+} from '@votingworks/ui';
+import { assert } from '@votingworks/utils';
+import { Loading } from './loading';
+import { AppContext } from '../contexts/app_context';
+
+function noop(): void {
+  // do nothing
+}
+
+interface NewPrintButtonProps extends StyledButtonProps {
+  print: () => Promise<void>;
+  title?: string;
+  showSimplePrintingModal?: boolean;
+  afterPrint?: () => void;
+  afterPrintError?: (errorMessage: string) => void;
+}
+
+export function NewPrintButton({
+  print,
+  title,
+  showSimplePrintingModal = true,
+  afterPrint,
+  afterPrintError,
+  children,
+  ...rest
+}: React.PropsWithChildren<NewPrintButtonProps>): JSX.Element {
+  const isMounted = useMountedState();
+  const makeCancelable = useCancelablePromise();
+  const { hasPrinterAttached } = useContext(AppContext);
+  const [isPrinting, setIsPrinting] = useState(false);
+
+  const needsPrinter = window.kiosk && !hasPrinterAttached;
+
+  const handlePrint = useCallback(async () => {
+    setIsPrinting(true);
+    setTimeout(() => {
+      if (!isMounted()) return;
+      setIsPrinting(false);
+    }, 3000);
```

Comment:
No idea, this is copied over code. @beausmith again, any idea?

Reply:
I'm not quite sure how the component could be no longer mounted and we need to change the component state. I will remove this.

Reply:
I think the `isMounted` check is likely to guard against the `act` warning in tests when you try to set state after unmount. But that doesn't explain what the whole timeout is for. (Based on your last comment, I think you may be misreading this - it executes `setIsPrinting(false)` if the component _is_ still mounted)

My read is that this timeout is stopping waiting for printing to complete after 3s, which closes the modal. My question was whether we need that timeout or if we can just wait for printing to complete before closing the modal.

Reply:
Ah, I was misreading it. The reason it is 3 seconds, in that case, is not because the print is expected to be done at that time. It's because most prints are longer and we're freeing up the UI while the print is probably still happening. We do this in the print button and at the end of printing the test deck. I think it's something we should keep.

Reply:
Ok, can you add a comment explaining this decision?